01-data-structures/07-graphs/kevin_bacon.rb

```diff
@@ -0,0 +1,121 @@
+require_relative "Node"
+
+$films = []
+$actors = []
+def find_kevin_bacon(node,start)
+  $actors << start.name
+
+  if start.name == "Kevin Bacon"
+    return $films
+  end
+
+  if $films.include?(:"Footloose")
+    $films = $films.uniq
```

Comment:
I get the sense that if you are going through duplicate movies, removing them actually makes the path inaccurate - If Bob is connected to James through That One Movie, James is connected to Bill through That Other Movie, Bill is connected to Lisa through What Is That Movie, Lisa is connected to Erica through What Is That Movie Again, Erica is connected to Jeffrey through How Many More Movies Are There, Jeffrey is connected to Lindsay That One Movie, and Lindsay is connected to Bacon through Finally Done With Movie Titles, then the path looks like so:

```
["That One Movie", "That Other Movie", "What Is That Movie", "What Is That Movie Again", "How Many More Movies Are There", "That One Movie", "Finally Done With Movie Titles"]
```
That path is longer than 6, and thus it should not be considered. Of course, the better path is `["That One Movie", "Finally Done With Movie Titles"]`, because Bob is actually connected to Lindsay directly through "That One Movie", but your algorithm does not know that. Anyway, removing the duplicates suddenly not only makes that a considered path, but it suggests that there is a direct connection from "How Many More Movies Are There" and "Finally Done With Movie Titles", which is completely invalid based on the actual path taken.

If you see anything wrong with my logic here, feel free to challenge my thinking. I am not always right, and it is good to get in the habit of checking those you feel are superior rather than blindly following them even if you think they may be wrong.